lib/structs.bzl

```diff
@@ -34,6 +34,22 @@ def _to_dict(s):
         if key != "to_json" and key != "to_proto"
```

Comment:
This method looks suspect: These functions no longer exist in the latest version of Bazel, but they may very well be used for other purposes and should be copied in that case.

Before adding a new function, could we make this more precise by checking the `type` of the values before skipping over these keys?

Reply:
> These functions no longer exist in the latest version of Bazel

They do not:

```
                print(struct().to_json())
Error: 'struct' value has no field or method 'to_json'
```

However, I cannot create a struct to test:

```
                structs.to_dict(struct(to_json = 1, to_proto = True)),
Error in struct: cannot override built-in struct function 'to_json'
```

I can implement the `type` check but won't be able to add a test.

Reply:
You can test with `last_green`: https://github.com/bazelbuild/bazel/commit/db5906870bdcb1631e1b18be8637aa21b79cf943

Reply:
https://github.com/bazel-contrib/bazel_features/pull/103